src/test/HostActivationTests/FrameworkResolution/ApplyPatchesSettings.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.FrameworkResolution
+{
+    public class ApplyPatchesSettings :
+        FrameworkResolutionBase,
+        IClassFixture<ApplyPatchesSettings.SharedTestState>
+    {
+        private SharedTestState SharedState { get; }
+
+        public ApplyPatchesSettings(SharedTestState sharedState)
+        {
+            SharedState = sharedState;
+        }
+
+        [Fact]
+        public void Default()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(MicrosoftNETCoreApp, "5.1.2"),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.3"));
+        }
+
+        [Fact]
+        public void RuntimeConfigOnly()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(MicrosoftNETCoreApp, "5.1.2")
+                    .WithApplyPatches(false),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.2"));
+        }
+
+        [Fact]
+        public void FrameworkReferenceOnly()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(new RuntimeConfig.Framework(MicrosoftNETCoreApp, "5.1.2")
+                        .WithApplyPatches(false)),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.2"));
+        }
+
+        [Fact]
+        public void Priority()
+        {
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(new RuntimeConfig.Framework(MicrosoftNETCoreApp, "5.1.2")
+                        .WithApplyPatches(false))
+                    .WithApplyPatches(true),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.2"));
+
+            RunTest(
+                runtimeConfig => runtimeConfig
+                    .WithFramework(new RuntimeConfig.Framework(MicrosoftNETCoreApp, "5.1.2")
+                        .WithApplyPatches(true))
+                    .WithApplyPatches(false),
+                result => result.Should().Pass()
+                    .And.HaveResolvedFrameworkVersion("5.1.3"));
+        }
+
+        [Fact]
+        public void InnerFrameworkReference_RuntimeConfig()
+        {
+            using (var dotnetCustomizer = SharedState.DotNetWithFrameworks.Customize())
+            {
+                dotnetCustomizer.Framework("MiddleWare").RuntimeConfig(runtimeConfig =>
+                    runtimeConfig.WithApplyPatches(false));
+
+                RunTest(
+                    runtimeConfig => runtimeConfig
+                        .WithFramework("MiddleWare", "2.1.0"),
+                    result => result.Should().Pass()
+                        .And.HaveResolvedFrameworkVersion("5.1.2")
```

Comment:
Are these version number transient? In the sense will they need to be changed as the framework and runtime evolves?

Reply:
The algorithm in `hostfxr` currently assigns no meaning to any particular version - just compares them one to the other.
Unless we change that behavior, the tests won't need to change either.